### PR TITLE
Make list of discoverers configurable

### PR DESCRIPTION
**Pull request by garuppel**

### CODE REVIEW

File: cmd/engine/engine.go [hunk 1]
```diff
@@ -196,6 +198,12 @@ func doCmd(_ *cobra.Command, _ []string) (err error) {
 		return fmt.Errorf("could not create storage: %w", err)
 	}
 
+	if viper.GetString(cli_discovery.DISCOVERER) == "" {
```
Comment (garuppel) [inline cmd/engine/engine.go:201]:
```suggestion
        // If no CSPs for discovering is given, take all implemented discoverers
	if viper.GetString(cli_discovery.DISCOVERER) == "" {
```

File: service/discovery/discovery.go [hunk 2]
```diff
@@ -100,6 +109,12 @@ func WithInternalAuthorizer(address string, username string, password string, op
 	}
 }
 
+func WithDiscoverers(cspList []string) ServiceOption {
```
Comment (garuppel) [inline service/discovery/discovery.go:112]:
```suggestion
// WithDiscoverers is an option to set CSPs for discovering
func WithDiscoverers(cspList []string) ServiceOption {
```

File: service/discovery/discovery.go [hunk 3]
```diff
@@ -154,12 +169,21 @@ func (s *Service) initAssessmentStream(additionalOpts ...grpc.DialOption) error
...
 // Start starts discovery
-func (s *Service) Start(_ context.Context, _ *discovery.StartDiscoveryRequest) (resp *discovery.StartDiscoveryResponse, err error) {
+func (s *Service) Start(_ context.Context, req *discovery.StartDiscoveryRequest) (resp *discovery.StartDiscoveryResponse, err error) {
 	resp = &discovery.StartDiscoveryResponse{Successful: true}
 
 	log.Infof("Starting discovery...")
 	s.scheduler.TagsUnique()
 
+	// Set CSPs
+	if req == nil || len(req.Csp) == 0 {
```
Comment (garuppel) [inline service/discovery/discovery.go:179]:
```suggestion
	if req == nil || req.Csp == nil || len(req.Csp) == 0 {
```

**Review by garuppel — COMMENTED** [review]
Some findings

File: api/discovery/discovery.proto [hunk 4]
```diff
@@ -55,7 +55,10 @@ service Discovery {
   }
 }
 
-message StartDiscoveryRequest {}
+message StartDiscoveryRequest {
+  // A list of CSPs to be discovered
+  repeated string csp = 1;
```
Comment (oxisto) [inline api/discovery/discovery.proto:60]: I would suggest `providers` as a property name.

File: cli/commands/service/discovery/discovery.go [hunk 5]
```diff
@@ -28,14 +28,20 @@ package discovery
...
 	"fmt"
+	"github.com/spf13/viper"
 
 	"clouditor.io/clouditor/api/discovery"
 	"clouditor.io/clouditor/cli"
 	"github.com/spf13/cobra"
 )
 
+const (
+	DISCOVERER = "discoverer"
```
Comment (oxisto) [inline cli/commands/service/discovery/discovery.go:39]: Uppercase? Should probably be named something like `DiscovererFlag`.

File: cmd/engine/engine.go [hunk 6]
```diff
@@ -106,6 +107,7 @@ var (
 	assessmentService    assessment.AssessmentServer
 	evidenceStoreService evidence.EvidenceStoreServer
 	db                   persistence.Storage
+	csps                 []string
```
Comment (oxisto) [inline cmd/engine/engine.go:110]:
```suggestion
	providers                 []string
```

File: service/discovery/discovery.go [hunk 7]
```diff
@@ -51,6 +52,12 @@ import (
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
+const (
+	CSPAWS   = "aws"
```
Comment (oxisto) [inline service/discovery/discovery.go:56]: ProviderAWS
ProviderK8S
ProviderAzure

File: service/discovery/discovery.go [hunk 8]
```diff
@@ -100,6 +109,13 @@ func WithInternalAuthorizer(address string, username string, password string, op
 	}
 }
 
+// WithDiscoverers is an option to set CSPs for discovering
+func WithDiscoverers(cspList []string) ServiceOption {
```
Comment (oxisto) [inline service/discovery/discovery.go:113]: WithProviders?

**Review by oxisto — COMMENTED** [review]
Very nice! Have been looking for this feature for a while now. Just some minor things like variable naming and style of constants; Looks good otherwise.

File: api/discovery/discovery.proto (hunk 4 above)
Comment (garuppel) [reply api/discovery/discovery.proto:60]: Perfect, that I was searching for...

**Review by oxisto — APPROVED** [review]

**Pull request merged**